instrumentation/jdbc/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jdbc/test/JdbcInstrumentationTest.java

```diff
@@ -0,0 +1,1387 @@
+import org.junit.jupiter.params.provider.ValueSource;
+
+class JdbcInstrumentationTest {
+
+  @RegisterExtension static final AutoCleanupExtension cleanup = AutoCleanupExtension.create();
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  private static String dbName;
+  private static String dbNameLower;
+  private static Map<String, String> jdbcUrls;
+  private static Map<String, String> jdbcDriverClassNames;
+  private static Map<String, String> jdbcUserNames;
+  private static Properties connectionProps;
+  // JDBC Connection pool name (i.e. HikariCP) -> Map<dbName, Datasource>
+  private static Map<String, Map<String, DataSource>> cpDatasources;
+
+  @BeforeAll
+  static void setUp() {
+    dbName = "jdbcUnitTest";
+    dbNameLower = dbName.toLowerCase(Locale.ROOT);
+    jdbcUrls =
+        ImmutableMap.of(
+            "h2", "jdbc:h2:mem:" + dbName,
+            "derby", "jdbc:derby:memory:" + dbName,
+            "hsqldb", "jdbc:hsqldb:mem:" + dbName);
+
+    jdbcDriverClassNames =
+        ImmutableMap.of(
+            "h2", "org.h2.Driver",
+            "derby", "org.apache.derby.jdbc.EmbeddedDriver",
+            "hsqldb", "org.hsqldb.jdbc.JDBCDriver");
+
+    jdbcUserNames = new HashMap<>();
+    jdbcUserNames.put("derby", "APP");
+    jdbcUserNames.put("h2", null);
+    jdbcUserNames.put("hsqldb", "SA");
+
+    connectionProps = new Properties();
+    connectionProps.put("databaseName", "someDb");
+    connectionProps.put("OPEN_NEW", "true"); // So H2 doesn't complain about username/password.
+
+    cpDatasources = new HashMap<>();
+
+    prepareConnectionPoolDatasources();
+  }
+
+  @AfterAll
+  static void tearDown() {
+    cpDatasources
+        .values()
+        .forEach(
+            k ->
+                k.values()
+                    .forEach(
+                        dataSource -> {
+                          if (dataSource instanceof Closeable) {
+                            try {
+                              ((Closeable) dataSource).close();
+                            } catch (IOException ignore) {
+                              // ignore
+                            }
+                          }
+                        }));
+  }
+
+  static void prepareConnectionPoolDatasources() {
+    List<String> connectionPoolNames = asList("tomcat", "hikari", "c3p0");
+    connectionPoolNames.forEach(
+        cpName -> {
+          Map<String, DataSource> dbDsMapping = new HashMap<>();
+          jdbcUrls.forEach(
+              (dbType, jdbcUrl) -> dbDsMapping.put(dbType, createDs(cpName, dbType, jdbcUrl)));
+          cpDatasources.put(cpName, dbDsMapping);
+        });
+  }
+
+  static DataSource createTomcatDs(String dbType, String jdbcUrl) {
+    org.apache.tomcat.jdbc.pool.DataSource ds = new org.apache.tomcat.jdbc.pool.DataSource();
+    String jdbcUrlToSet = Objects.equals(dbType, "derby") ? jdbcUrl + ";create=true" : jdbcUrl;
```

Comment:
the benefit of `Objects.equals()` is that it ensures that if connectionPoolName is null, no NullPointerException will be thrown. The method internally checks if either argument is null and handles it gracefully. In our case within this test, we don't expect it to be null, so we could simplify these expressions

```suggestion
    String jdbcUrlToSet = dbType.equals("derby") ? jdbcUrl + ";create=true" : jdbcUrl;
```